### ISSUE TITLE
Turn off rate limit for cron monitoring

### ISSUE THREAD

**Issue by phips28** [getsentry/sentry#80502]
### Problem Statement

How can I turn off the rate limit for cron monitoring on a self-hosted instance?

**6 check-ins per minute** is just useless for us. 🙁 

https://docs.sentry.io/platforms/javascript/guides/node/crons/#rate-limits
> Crons imposes a rate limit on check-ins to prevent abuse and resource overuse. Specifically, you can only send a maximum of **6 check-ins per minute** per existing monitor environment. This limit is enforced on a per-project basis, meaning that the rate limit applies collectively to all monitor environments within a given project. You can check if any of your check-ins are being dropped in the [Usage Stats](https://docs.sentry.io/product/stats/#usage-stats) page.

**Comment by phips28** [comment]
we also need this!

**Comment by wedamija** [comment]
We can look into making this configurable, but I'm interested in what your case is here? Crons might not work all that consistently if you're sending a lot of checks per monitor environment every minute

**Comment by phips28** [comment]
we have crons in the seconds area (e.g. every 5 sec) - small high frequent tasks.

**Comment by evanpurkhiser** [comment]
If you're able to patch your sentry instance you would want to change this

https://github.com/getsentry/sentry/blob/0ee4e5d56b7837b76bf26f623fa99dcd176f873f/src/sentry/monitors/rate_limit.py#L49

**Comment by evanpurkhiser** [comment]
> we have crons in the seconds area (e.g. every 5 sec) - small high frequent tasks.

I'm sure you're already aware, but fwiw Crons won't detect misses at this granularity. It will of course report failures, which I'm guessing is what you're after right?

**Comment by phips28** [comment]
> > we have crons in the seconds area (e.g. every 5 sec) - small high frequent tasks.
> 
> I'm sure you're already aware, but fwiw Crons won't detect misses at this granularity. It will of course report failures, which I'm guessing is what you're after right?

yes I am aware of this, and I only want to detect a miss when there was no heartbeat after 5min for example.

**Comment by phips28** [comment]
> If you're able to patch your sentry instance you would want to change this
> 
> [sentry/src/sentry/monitors/rate_limit.py](https://github.com/getsentry/sentry/blob/0ee4e5d56b7837b76bf26f623fa99dcd176f873f/src/sentry/monitors/rate_limit.py#L49)
> 
> Line 49 in [0ee4e5d](/getsentry/sentry/commit/0ee4e5d56b7837b76bf26f623fa99dcd176f873f)
> 
>  ALLOWED_CHECK_INS_PER_MONITOR = 6

would be nice if this would be an ENV var / configurable.

**Comment by evanpurkhiser** [comment]
sorry for the late reply here, but for now this is on our backlog! PRs of course always welcome!

**Comment by niklasHack2810** [comment]
We are also hitting this limit, is there any workaround by now? We use docker so i dont think we can easily alter code in sentry.

**Comment by evanpurkhiser** [comment]
Adding a new sentry option `crons.per_monitor_rate_limit` to control this :)

You'll be able to set it in your `~/.sentry/config.yml` file (or in `sentry.conf.py` as `SENTRY_OPTIONS['crons.per_monitor_rate_limit']`

**Comment by phips28** [comment]
When?

**Comment by wedamija** [comment]
> When?

It will be available in the next self hosted release. These are released on a regular schedule around the 15th of each month.

**Comment by rodolfoBee** [comment]
Reopening this issue.
A self-hosted user on v25.6.1 reported:
> Based on https://github.com/getsentry/sentry/issues/80502#issuecomment-2691699164 we should be able to change the cron monitoring rate limit here. however notice it still does not work and still gets A sent check-in was dropped due to being rate limited. Reivew our rate limits for more information., suspect due to https://github.com/getsentry/sentry/blob/master/src/sentry/monitors/consumers/monitor_consumer.py#L85 ?

**Comment by gaprl** [comment]
Hey folks, we have a fix for this on https://github.com/getsentry/sentry/pull/95743 and should be able to merge early next week.
